scripts/optimize_bayesopt.py

```diff
@@ -107,10 +120,11 @@ def target_func(localAreaDensity):
     return score
 
 
-def optimize_local_area_density():
+def optimize():
     # define bounds for the params you want to optimize. Can be multivariate. Check https://github.com/fmfn/BayesianOptimization on how to
```

Comment:
I think the main challenge is that Bayes Opt requires bounds to be set while PSO as well as GridSearch only require a "starting point". I think such a change would require a change in the framework itself to be able to add bounds instead of the default_parameter dict that is currently used.

Reply:
Plus I like the idea to have on example for use with the htm.core opt framework and one for "bring your own" opt.